Sprint 13

ntqoglu opened the pull request.

ntqoglu committed:
Sprint 13. PlayerActivity layout

ntqoglu committed:
Sprint 13. backup

ntqoglu committed:
Sprint 13. PlayerActivity works

ntqoglu committed:
Sprint 13. PlayerActivity.xml works in dark mode

ntqoglu committed:
Sprint 13. Final

kekulta commented on app/src/main/java/com/example/playlistmaker/PlayerActivity.kt:
```diff
@@ -0,0 +1,64 @@
+
+class PlayerActivity : AppCompatActivity() {
+
+    private lateinit var binding: ActivityPlayerBinding
```
Молодец, что испольщуешь байндинг!

kekulta commented on app/build.gradle.kts:
```diff
@@ -1,6 +1,7 @@
 plugins {
     alias(libs.plugins.android.application)
     alias(libs.plugins.jetbrains.kotlin.android)
+    id("kotlin-parcelize")
```
Если используешь version catalog, то лучше использовать его везде.

kekulta commented on app/src/main/java/com/example/playlistmaker/PlayerActivity.kt:
```diff
@@ -0,0 +1,64 @@
+
+        binding.tbPlayer.setNavigationOnClickListener { finish() }
+
+        val track = intent.getParcelableExtra<Track>("track")
```
Этот метод депрекейтед. Можно использовать IntentCompat.

kekulta commented on app/src/main/java/com/example/playlistmaker/TrackAdapter.kt:
```diff
@@ -25,10 +26,26 @@ class TrackAdapter(
+
+            val context = holder.itemView.context
+            val intent = Intent(context, PlayerActivity::class.java).apply {
+                putExtra("track", track)
+                putExtra("trackId", track.trackId)
+                putExtra("trackName", track.trackName)
+                putExtra("artistName", track.artistName)
+                putExtra("trackTimeMillis", track.trackTimeMillis)
+                putExtra("artworkUrl100", track.artworkUrl100)
+                putExtra("trackCountry", track.trackCountry)
+                putExtra("trackAlbum", track.trackAlbum)
+                putExtra("genre", track.genre)
+                putExtra("releaseDate", track.releaseDate)
```
Не вижу смысла класть парселабл и одновременно  разбивать отдельно по полям.

kekulta commented on app/src/main/java/com/example/playlistmaker/models/Track.kt:
```diff
@@ -1,11 +1,18 @@
     @SerializedName("trackTimeMillis") val trackTimeMillis: Int,
-    @SerializedName("artworkUrl100") val artworkUrl100: String
-)
+    @SerializedName("artworkUrl100") val artworkUrl100: String,
+    @SerializedName("country") val trackCountry: String,
+    @SerializedName("collectionName") val trackAlbum: String,
+    @SerializedName("primaryGenreName") val genre: String,
+    @SerializedName("releaseDate") val releaseDate: String
```
Молодец, что подписал все поля как @SerializedName

ntqoglu committed:
Sprint 13. Final. Fix:
Углы обложки трека на экране плеера закруглены.
Есть таймкода 00:00.
Если набрать запрос, который не даст результато и затем стереть с клавиатуры, то получим корректный экран.

ntqoglu committed:
Sprint 13. Final. Fix 2

ntqogly commented:
Исправил все отмеченные замечания. Остальные, твои рекомендации, обязательно поправлю как только можно будет закрыть спринт. Спасибо тебе огромное, что отметил их! Надеюсь работа будет принята..

ntqoglu committed:
Merge branch 'master' into Sprint-13
